akka-stream/src/main/scala/akka/stream/impl/fusing/FutureFlow.scala

```diff
@@ -0,0 +1,145 @@
+        override def onDownstreamFinish(cause: Throwable): Unit =
+          if (delayDownstreamCancellation) {
+            downstreamCause = OptionVal.Some(cause)
+          } else {
+            innerMatValue.failure(new NeverMaterializedException(cause))
+            super.onDownstreamFinish(cause)
+          }
+
+        def onFuture(futureRes: Try[Flow[In, Out, M]]) = futureRes match {
+          case Failure(exception) =>
+            setKeepGoing(false)
+            innerMatValue.failure(new NeverMaterializedException(exception))
+            failStage(exception)
+          case Success(flow) =>
+            //materialize flow, connect inlet and outlet, feed with potential events and set handlers
+            connect(flow)
+            setKeepGoing(false)
+        }
+      }
+      def connect(flow: Flow[In, Out, M]): Unit = {
+        val subSource = new SubSourceOutlet[In](s"${FutureFlow.this}.subIn")
+        val subSink = new SubSinkInlet[Out](s"${FutureFlow.this}.subOut")
+
+        subSource.setHandler {
+          new OutHandler {
+            override def onPull(): Unit = if (!isClosed(in)) tryPull(in)
+
+            override def onDownstreamFinish(cause: Throwable): Unit = if (!isClosed(in)) cancel(in, cause)
+          }
+        }
+        subSink.setHandler {
+          new InHandler {
+            override def onPush(): Unit = push(out, subSink.grab())
+
+            override def onUpstreamFinish(): Unit = complete(out)
+
+            override def onUpstreamFailure(ex: Throwable): Unit = fail(out, ex)
+          }
+        }
+        Try {
+          Source.fromGraph(subSource.source).viaMat(flow)(Keep.right).to(subSink.sink).run()(subFusingMaterializer)
+        } match {
+          case Success(matVal) =>
+            innerMatValue.success(matVal)
+            Initializing.upstreamFailure match {
+              case OptionVal.Some(ex) =>
+                subSource.fail(ex)
+              case OptionVal.None =>
+                if (isClosed(in))
+                  subSource.complete()
+            }
+            Initializing.downstreamCause match {
+              case OptionVal.Some(cause) =>
+                subSink.cancel(cause)
+              case OptionVal.None =>
+                //todo: should this be invoked before and independently of checking downstreamCause?
+                // in most case if downstream pulls and then closes, the pull is 'lost'. is it possible for some flows to actually care about this? (non-eager broadcast?)
+                if (isAvailable(out)) {
+                  subSink.pull()
+                }
+            }
+          case Failure(ex) =>
+            innerMatValue.failure(new NeverMaterializedException(ex))
+            failStage(ex)
+        }
+        setHandlers(in, out, new InHandler with OutHandler {
+          override def onPull(): Unit = subSink.pull()
+          override def onDownstreamFinish(cause: Throwable): Unit = subSink.cancel(cause)
```

Comment:
1. this is the old behaviour (I think :sunglasses:)
2. once materialized the nested flow takes charge of the stage's ports so it makes sense it should react to downstream cancellation.

the way I see it, the attribute controls what happens before nested flow's materialization, once materialized both modes converge into one.

Reply:
if we don't propagate cancellation the nested flow will never stop which as far as I understand things will prevent the interpreter from stopping (since it'd still have running graph stages).
aside from this, this may change the nested flow behaviour and materialized value, i.e. a flow that uses `alsoToMat` in order to aggregate and report metrics/whatever will never complete its matVal if it doesn't receive a downstream cancellation signal.

Reply:
Yep, think it's fine as it is now.